cml/setup.sh

```diff
@@ -21,11 +21,13 @@ curl -sL https://deb.nodesource.com/setup_12.x | sudo bash
 sudo apt update && sudo apt-get install -y nodejs
 
 sudo apt install -y ubuntu-drivers-common git
-sudo ubuntu-drivers autoinstall 
+sudo ubuntu-drivers autoinstall
 
-curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey | sudo apt-key add - 
+curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey | sudo apt-key add -
 curl -s -L https://nvidia.github.io/nvidia-docker/ubuntu18.04/nvidia-docker.list | sudo tee /etc/apt/sources.list.d/nvidia-docker.list
-sudo apt update && sudo apt install -y nvidia-docker2
+sudo apt update && sudo apt install -y nvidia-docker2 jq moreutils
+
+cat /etc/docker/daemon.json | jq '.+{"default-runtime": "nvidia"}' | sudo sponge /etc/docker/daemon.json
```

Comment:
> Changes introduced with this pull request cause GPU-enabled container images to malfunction when used with:
> 
> 1. Non-GPU AWS machines
> 2. Jobs that don't include the --gpus option?
>
> https://github.com/iterative/cml/issues/611#issuecomment-873159007

***

Solving **(1)** should be as easy as setting the default runtime only if there is a GPU installed:

```suggestion
nvidia-smi && cat /etc/docker/daemon.json | jq '.+{"default-runtime": "nvidia"}' | sudo sponge /etc/docker/daemon.json
```

Unfortunately, **(2)** doesn't seem to have an easy solution.